feat(app): use new TimeAverageToggle

VIKTORVAV99 opened the pull request:
## Issue

Part of: AVO-357

## Description

Uses the new TimeAverage toggle of #6911 to make the PR easier to review and reduce the scope of the other PR.

It also removes the last icon we used from FontAwesome as the icon is no longer in the design.

### Preview
<img width="451" alt="image" src="https://github.com/user-attachments/assets/3140cbd6-9be4-4cd9-9dda-9eb37e6a3a0e">


### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

Alportan commented:
Looks good from a design perspective ✅  Hurray 🎉 

One question, when I switch between the toggle items, I see a "Data is temporary unavailable" message. Is that because of dev environment?

![Screen Cast 2024-08-28 at 3 06 47 PM](https://github.com/user-attachments/assets/75308840-14b4-4244-8ab2-8b00adc36787)

VIKTORVAV99 commented:
> Looks good from a design perspective ✅ Hurray 🎉
> 
> One question, when I switch between the toggle items, I see a "Data is temporary unavailable" message. Is that because of dev environment? I tried a gif, was hard to see due to it's frame rate, but I added a screenshot instead 😅
> 
> ![Screen Cast 2024-08-28 at 3 06 47 PM](https://private-user-images.githubusercontent.com/149581095/362293134-bc57bd06-85bc-42b1-8ca3-968ceaa8f38d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4NTU5MjksIm5iZiI6MTcyNDg1NTYyOSwicGF0aCI6Ii8xNDk1ODEwOTUvMzYyMjkzMTM0LWJjNTdiZDA2LTg1YmMtNDJiMS04Y2EzLTk2OGNlYWE4ZjM4ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOFQxNDMzNDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZTkzMDdiZjdhZTIyNzgwZjM3MGM3YmRkOGI4OWYwMGZlYTc3YTQ3M2JmMzFmMWQyM2RmZGViYWM4MjQyNGI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3TbhSrnjZjWpECPJAEfzRk7X1nyDLjHScsmw8iH7N6A)

Not because of the dev environment but we have a nasty bug somewhere in the time control logic where a useEffect hook is triggering rendering twice that is proving hard to fix without breaking other stuff.

This is why I added the suspense query before as it should in theory at least hide the issue but that made the whole time controller flash instead which was also sub optimal. What I think I'm gonna try next is creating a simple skeleton like we talked about or some form of hybrid so only parts of the controller are affected.

madsnedergaard commented on web/src/components/TimeAverageToggle.tsx:
```diff
@@ -17,42 +20,41 @@ export interface TimeAverageToggleProps {
 
   return (
-    <ToggleGroupPrimitive.Root
+    <ToggleGroupRoot
```
Would be nice to add a keyboard focus state, seems like it worked better before (not sure why though) :)

Alportan replied:
Good catch, we can go with a simple outline of the pill as an easy fix 🙌 

![image](https://github.com/user-attachments/assets/d1a4f129-1d80-44ad-b1bd-089d9d349229)

VIKTORVAV99 replied:
Are you adding the colors to the Figma or is it the brand color now (but with the dark variant as well)?

Alportan replied:
Brand color, but[ added it to Figma](https://www.figma.com/design/O1DtS6UyZHSH4IymG8Ufsw/%F0%9F%8E%B9-Design-Hack?node-id=5257-5632&t=ntt7JkKDAaO1ylDA-1) as well! 🙌 My bad, I didn't link it before 🙈

madsnedergaard replied:
`shift+tab` two times on a refreshed view takes you to it, then you can go sideways with arrows - so I think it would be nice to just quickly a `focus:outline-brand-green` kinda class as a first step :)

VIKTORVAV99 replied:
Yeah I think that would work!

I'll add it then!

madsnedergaard commented on web/src/features/time/TimeControllerWrapper.tsx:
```diff
@@ -32,7 +32,7 @@ function BottomSheetWrappedTimeController() {
       header={<TimeHeader />}
     >
-      <TimeController className="p-2 pt-1 min-[370px]:px-4" />
+      <TimeController className="p-2 min-[370px]:px-4" />
```
On smaller screens (and languages with longer text) it would be nice to have some top padding again:

**Before:**


![Screenshot 2024-08-30 at 14 55 23](https://github.com/user-attachments/assets/704d44bd-2d20-472e-b586-626a8e14e47b)


**This PR:**

![Screenshot 2024-08-30 at 14 55 17](https://github.com/user-attachments/assets/f0b29631-926d-498d-a230-046462a73b28)

Alportan replied:
Good catch Mads! 🙌 Some extra padding around the header would be nice 🫶

VIKTORVAV99 replied:
The padding itself is actually doubled but I think it's the overflow that's the problem by setting a fixed hight of the "title" to avoid CLS caused by the badge.

But I'll try out a few solutions to see what is working best 🙂

madsnedergaard reviewed (approved):
LGTM 🥳
